mono/metadata/appdomain.c

```
@@ -223,14 +223,22 @@ create_domain_objects (MonoDomain *domain)
  *
  */
 void
-mono_runtime_init (MonoDomain *domain, MonoThreadStartCB start_cb,
-		   MonoThreadAttachCB attach_cb)
+mono_runtime_init (MonoDomain *domain, MonoThreadStartCB start_cb, MonoThreadAttachCB attach_cb)
 {
 	MonoError error;
+	mono_runtime_init_checked (domain, start_cb, attach_cb, &error);
+	mono_error_raise_exception (&error);
```

This is a runtime API, it can't raise.
Just discard the error and ban it.